### PR TITLE
tbcheck ussd update

### PR DESCRIPTION
**Pull request by Buhle79**

### CODE REVIEW

File: src/ussd_tb_check.js [hunk 1]
```diff
@@ -660,7 +660,12 @@ go.app = function () {
       return new JsonApi(self.im)
         .post(self.im.config.healthcheck.url + "/v2/tbcheck/", payload)
         .then(
-          function () {
+          function (response) {
+            self.im.user.answers = {
```
Comment (erikh360) [inline src/ussd_tb_check.js]: Won't his overwrite the whole answers object with a new one? I think it's better to set each value separately

File: src/ussd_tb_check.js [hunk 2]
```diff
@@ -691,40 +695,226 @@ go.app = function () {
         question: text,
         error: error,
         accept_labels: true,
-        choices: [new Choice("state_show_results", $("See Results"))],
+        choices: [new Choice("state_display_arm_message", $("See Results"))],
       });
     });
 
-    self.states.add("state_show_results", function (name) {
```
Comment (erikh360) [inline src/ussd_tb_check.js:698]: Shouldn't we still show the original message if study is not active yet and arm is empty?

File: src/ussd_tb_check.js [hunk 3]
```diff
@@ -691,40 +695,226 @@ go.app = function () {
...
+    self.states.add("state_display_arm_message", function (name) {
       var answers = self.im.user.answers;
-      var risk = self.calculate_risk();
-      var text = $(
-        "You don't need a TB test now, but if you develop cough, fever, weight loss " +
-          "or night sweats visit your nearest clinic."
-      );
+      var arm = answers.group_arm;
+      return self.states.create("state_" + arm);
```
Comment (erikh360) [inline src/ussd_tb_check.js]: This can be one line if we only use the arm in the one place.

**Review by erikh360 — COMMENTED** [review]
Looks good so far, just a few comments.

Please also make sure you have tests around all the new states to see that the flow is correct.

File: src/ussd_tb_check.js (hunk 2 above)
Comment (Buhle79) [reply src/ussd_tb_check.js:698]: @erikh360 Which one is the original message ?

File: src/ussd_tb_check.js (hunk 1 above)
Comment (Buhle79) [reply src/ussd_tb_check.js]: Oh yes it does override it thanks.

File: src/ussd_tb_check.js [hunk 4]
```diff
@@ -660,7 +660,14 @@ go.app = function () {
       return new JsonApi(self.im)
         .post(self.im.config.healthcheck.url + "/v2/tbcheck/", payload)
         .then(
-          function () {
+          function (response) {
+            answers.group_arm = response.data.tbconnect_group_arm.toLowerCase();
+            answers.tbcheck_id = response.data.id;
+            /*self.im.user.answers = {
```
Comment (erikh360) [inline src/ussd_tb_check.js]: Can we remove this completely please.

**Review by erikh360 — APPROVED** [review]
just the one small comment

**Pull request merged**